modules/account-lib/src/coin/stx/transaction.ts

```diff
@@ -0,0 +1,171 @@
+import BigNumber from 'bignumber.js';
+import {
+  BufferReader,
+  PayloadType,
+  StacksTransaction,
+  TransactionSigner,
+  createStacksPrivateKey,
+  deserializeTransaction,
+  addressToString,
+  StacksMessageType,
+  createStacksPublicKey,
+  SingleSigSpendingCondition,
+  isSingleSig,
+  StacksPublicKey,
+  TransactionAuthField
+} from '@stacks/transactions';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { SigningError } from '../baseCoin/errors';
+import { BaseKey } from '../baseCoin/iface';
+import { BaseTransaction, TransactionType } from '../baseCoin';
+import { SignatureData, TxData } from './iface';
+import { getTxSenderAddress, bufferToHexPrefixString, removeHexPrefix } from './utils';
+import { KeyPair } from './keyPair';
+
+export class Transaction extends BaseTransaction {
+  private _stxTransaction: StacksTransaction;
+  protected _type: TransactionType;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  canSign(key: BaseKey): boolean {
+    // TODO: implement this validation method
+    return true;
+  }
+
+  async sign(keyPair: KeyPair[] | KeyPair): Promise<void> {
+    const keyPairs = keyPair instanceof Array ? keyPair : [keyPair];
+    const signer = new TransactionSigner(this._stxTransaction);
+    for (const kp of keyPairs) {
+      const keys = kp.getKeys(true);
+      if (!keys.prv) {
+        throw new SigningError('Missing private key');
+      }
+      const privKey = createStacksPrivateKey(keys.prv);
+      signer.signOrigin(privKey);
```

Comment:
Do we not save any state in this method? We don't set _stxTransaction or any signature data or anything like that?

Reply:
The signer signs the transaction (updates spendingCondition in transaction authorization)

Reply:
got it, thanks